components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -33,6 +35,19 @@ import { Store } from '../../common/async/types'
 import { HardwareWalletAccount } from 'components/brave_wallet_ui/common/hardware/types'
 import { GetTokenParam } from '../../utils/api-utils'
 
+function switchEndianness (hexString: string) {
+  const regex = hexString.match(/.{2}/g)
+  if (!regex) {
+    throw new Error(`Could not switch endianness of hex string: ${hexString}`)
+  }
+  return regex.reverse().join('')
+}
+
+function extractPublicKeyForBLS (privateKey: string): string {
+  // https://github.com/brave/brave-browser/issues/20024
+  return Buffer.from(bls.getPublicKey(switchEndianness(privateKey))).toString('hex')
+}
```

Comment:
We should move such functions out of this file to some place like `components/brave_wallet_ui/utils/crypto-utils.ts`. Similarly, `switchEndianness` would go in `hex-utils.ts`.

Reply:
I will move them to filecoin keyring in next PR